graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/JavaPoetUtils.kt

```diff
@@ -207,21 +209,30 @@ fun customAnnotation(annotationArgumentMap: MutableMap<String, Value<Value<*>>>,
 /**
  * Generates the code block containing the parameters of an annotation in the format value
  */
-private fun generateCode(value: Value<Value<*>>, packageName: String = ""): CodeBlock =
+private fun generateCode(config: CodeGenConfig, value: Value<Value<*>>, annotationName: String, prefix: String = ""): CodeBlock =
     when (value) {
         is BooleanValue -> CodeBlock.of("\$L", (value as BooleanValue).isValue)
         is IntValue -> CodeBlock.of("\$L", (value as IntValue).value)
-        is StringValue -> CodeBlock.of("\$S", (value as StringValue).value)
+        is StringValue ->
+            // If string ends with .class, treat as class object
+            if ((value as StringValue).value.takeLast(6) == ".class") {
+                val className = (value as StringValue).value.dropLast(6)
```

Comment:
Maybe create a constant for 6 and have comment to show. what it is

Reply:
Thinking about it more. It maybe better to have constants for .class and ::class. and take its length instead of 6 or 7.

Reply:
I have thought about it a bit more. I think we should have the same way in graphql to identify a class irrespective of if we are generating java class or Kotlin. Coz if we are using the same graphql for 2 different service, one generating java and one kotlin, we wont be able to do it if we have different representations for class. Let me know if you cant understand.

Reply:
makes sense

Reply:
Done